CONTRIBUTING.md

```diff
@@ -0,0 +1,227 @@
+# Contributing
+
+Thanks for considering contributing to Subspace Network!
+
+The best way to start is to join our [Forum](https://forum.subspace.network/) or
+[Discord](https://discord.gg/subspace-network), [official website](https://subspace.network/) has a bunch of learning
+materials too.
+
+GitHub is primarily used by developers to work on new features, improvements and fix bugs. If you're not sure whether
+you have an actual bug that needs to be fixed or you just want to ask a question, forum is the best place for that.
+
+This document describes expectations and recommendations that should make contributions such as issues and pull requests
+on GitHub productive for everyone involved.
+
+## Process optimized for maintainers
+
+Subspace Network is maintained by a small group of passionate individuals with limited time. Number of potential outside
+contributors is much larger than number of maintainers and wider community of users is even bigger than that.
+
+As such, the process of contributing to Subspace Network needs to be optimized in such a way that it doesn't overwhelm
+maintainers with unnecessary notifications, questions that can be answered by wider community and follow practices
+defined in this document below, such that contributions can be processed in the most productive way saving time and
+making it a better experience overall.
+
+## Bugs, improvements and feature requests
+
+These should be submitted in such a way that they are actionable by developers. Again, if you're unsure or just have a
+question - use forum, GitHub issue can be created later if it is necessary.
+
+Always search for similar topics already created on the forum or here on GitHub before creating a new one, but don't
+hijack threads that looks somewhat similar, but are not necessarily the same or have the same root cause.
+
+### Bugs
+
+For bug report to be actionable it should ideally follow the following template:
```

Comment:
[Opened a PR](https://github.com/subspace/subspace/pull/734) to add a default bug report template provided by GitHub for those that open directly on issues.